tardis/adapters/sites/htcondor.py

```diff
@@ -244,14 +244,18 @@ async def deploy_resource(
             resource_attributes.obs_machine_meta_data_translation_mapping,
         )
 
+        def job_environment(seperator):
+            return seperator.join(
+                f"TardisDrone{key}={value}" for key, value in drone_environment.items()
+            )
+
         submit_jdl = jdl_template.substitute(
             machine_meta_data_translation(
                 self.machine_meta_data,
                 self.htcondor_machine_meta_data_translation_mapping,
             ),
-            Environment=";".join(
-                f"TardisDrone{key}={value}" for key, value in drone_environment.items()
-            ),
+            Environment=job_environment(";"),
+            Arguments=job_environment(" "),
```

Comment:
So now both values are taken? Is `Environment` still used until this is deprecated or will this be used like this from now on? I thought initially that the relevant information is taken now solely from `Arguments`. 

Currently it means, that the same information are put twice. So there is a chance for failure. What I currently don't see at all is where `Arguments` are read 😬 and how the use of `Environment` does change for drones.

Reply:
Some htcondor instances, almost all htcondor-ces remove the `Environment`. The usage of `Arguments` is similar to what the WLCG pilot jobs do. I think we can drop the ` Environment`  but it should be there until we announced it as deprecated.

Reply:
Both template parameters`Environment` and `Arguments` can be used in the JDL template. A deprecation of `Environment` is not forseen. The information is the same except the separator, which is due to the different syntax used in the HTCondor JDL. 
We are using Python [String Templates](https://docs.python.org/3/library/string.html?highlight=template#string.Template) here. We can define arbitary template parameter, but not all of them need to be used in the HTCondor JDL template. As stated in the [documentation](https://cobald-tardis.readthedocs.io/en/latest/adapters/site.html#htcondor-site-adapter), an example is 

```
executable = start_pilot.sh
transfer_input_files = setup_pilot.sh,grid-mapfile
output = logs/$$(cluster).$$(process).out
error = logs/$$(cluster).$$(process).err
log = logs/cluster.log

accounting_group=tardis
x509userproxy = /home/tardis/proxy

environment=${Environment}

request_cpus=${Cores}
request_memory=${Memory}
request_disk=${Disk}
```

All ${...} will be replaced with template parameters set in `TARDIS`. Now, there is also the possibility to use the `Arguments` template parameter like in 

```
executable = start_pilot.sh
arguments = ${Arguments}
transfer_input_files = setup_pilot.sh,grid-mapfile
output = logs/$$(cluster).$$(process).out
error = logs/$$(cluster).$$(process).err
log = logs/cluster.log

accounting_group=tardis
x509userproxy = /home/tardis/proxy

request_cpus=${Cores}
request_memory=${Memory}
request_disk=${Disk}
```

or even both of them.

In out case, we keep on using `Environment` using HTCondors `VanilaUniverse` and use `Arguments` on the HTCondor `GridUniverse`.  The drone in the latter case is then just doing a

```
#set environment from arguments
for arg in $@; do
  eval export $arg
done
```

for example. I will update the documentation accordingly.

Reply:
Having the adapted documentation is great 👍 
However, I still see that this might cause problems because we are with this introducing two methods how the environment is actually being set. For the VanillaUniverse it is by using `Environment` and for the GridUniverse it is `Arguments`. 
I vote for deprecating the use of `Environment` as this would only leave one valid way of configuring the environment of drones and therefore might help with maintainability and might even reduce misconfigurations. Or is there a enforcing reason for still going with `Environment`?

Reply:
In my opionion `Environment` is the acutal way to set environment variables for the job execution. Using `Arguments` is just a workaround to make it work with Grid CEs droping the environment set by the user. So, it would be more natural to keep `Environment` and have `Arguments` in additon.